### PR TITLE
Attach tf version to the User-Agent header

### PR DESCRIPTION
**Pull request by nvthongswansea**
- Implement a part of #108.
- Attach tf version to the User-Agent header.

### CODE REVIEW

File: gridscale/provider.go [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 
 import (
+	"runtime"
 	"strings"
 
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/terraform"
 )
 
+const version = "1.8.1"
```
Comment (bkircher) [inline gridscale/provider.go]: Hmm, yeah. Might work for now. But we can get that thing from goreleaser, I think. Right?

```
builds:
    ldflags:
      - "-s -w -X github.com/gridscale/terraform-provider-gridscale/somePackage.GitCommit={{.Commit}} -X github.com/gridscale/terraform-provider-gridscale/somePackage.Version={{.Version}}"
```

This way it always works including "dirty" builds.

**Review by bkircher — CHANGES_REQUESTED** [review]

**Comment by nvthongswansea** [comment]
@bkircher  test build command 
```
go install -X main.version=testver -X main.commit=testcommit -X github.com/terraform-providers/terraform-provider-gridscale/gridscale.version=testver  -X github.com/terraform-providers/terraform-provider-gridscale/gridscale.commit=testcommit
``` 
Run `terraform init`, and then run `TF_LOG=DEBUG terraform apply`.

**Comment by bkircher** [comment]
I just tried this for testing but to no avail. Terraform implements some sort of TOFU model and remembers the hash of the TF provider binary in the dependency lock file

So this does not work:

- Build with `goreleaser build --snapshot --rm-dist`; this will create a binary in `dist/terraform-provider-gridscale_linux_amd64/`
- Go to a test TF project
- Do `terraform init`
- cp over our previously build binary
- run `terraform plan`
Any ideas?

**Comment by bkircher** [comment]
Ah, just read your comment.

**Comment by nvthongswansea** [comment]
> I just tried this for testing but to no avail. Terraform implements some sort of TOFU model and remembers the hash of the TF provider binary in the dependency lock file
> 
> So this does not work:
> 
> * Build with `goreleaser build --snapshot --rm-dist`; this will create a binary in `dist/terraform-provider-gridscale_linux_amd64/`
> * Go to a test TF project
> * Do `terraform init`
> * cp over our previously build binary
> * run `terraform plan`
>   Any ideas?

Just did what you told above. It works for me. The only extra steps are:
1. Change the name of `terraform-provider-gridscale_vv1.8.0-SNAPSHOT-...`binary file to `terraform-provider-gridscale`. Move this binary file to `.../go/bin/`.
2. Change the permission of the file, o.w permission denied error will happen (or it may just happen on my WSL).
3. Do your steps.

**Comment by nvthongswansea** [comment]
@bkircher This is what I've got 
image

**Comment by bkircher** [comment]
Got it. LGTM

**Pull request merged**